--- ../setup-env.sh.orig	2019-05-22 08:37:29.759969941 +0200
+++ ./releng/setup-env.sh	2019-05-22 08:36:16.229970487 +0200
@@ -97,8 +97,8 @@
 fi
 
 if [ $host_platform = android ]; then
-  ndk_required_name=r17b
-  ndk_required_version=17.1.4828580
+  ndk_required_name=r19b
+  ndk_required_version=19.1.5304403
   if [ -n "$ANDROID_NDK_ROOT" ]; then
     if [ -f "$ANDROID_NDK_ROOT/source.properties" ]; then
       ndk_installed_version=$(grep Pkg.Revision "$ANDROID_NDK_ROOT/source.properties" | awk '{ print $NF; }')
@@ -465,7 +465,7 @@
 
     case $host_arch in
       x86)
-        android_api=18
+        android_api=21
         host_triplet="i686-linux-android"
         host_arch_flags="-march=i686"
         host_ldflags="-fuse-ld=gold"
@@ -477,7 +477,7 @@
         host_ldflags="-fuse-ld=gold -Wl,--icf=all"
         ;;
       arm)
-        android_api=18
+        android_api=21
         host_triplet="arm-linux-androideabi"
         host_arch_flags="-march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16"
         host_ldflags="-fuse-ld=gold -Wl,--icf=all -Wl,--fix-cortex-a8"
