diff -Naur hping3-20051105/Makefile hping3-3.0.0/Makefile
--- hping3-20051105/Makefile	2005-11-05 16:01:41.000000000 +0530
+++ hping3-3.0.0/Makefile	2018-02-22 18:27:39.000000000 +0530
@@ -6,15 +6,16 @@
 # $date: Sun Jul 25 17:56:15 MET DST 1999$ 
 # $rev: 3$ 
 
-CC= gcc
-AR=/usr/bin/ar
-RANLIB=/usr/bin/ranlib
-CCOPT= -O2 -Wall  -I/usr/include/tcl8.4 -DUSE_TCL
+AR= $AR
+LD= $LD
+CC= $CC
+RANLIB= $RANLIB
+CCOPT= -O2 -Wall   -DUSE_TCL
 DEBUG= -g
 #uncomment the following if you need libpcap based build under linux
 #(not raccomanded)
 COMPILE_TIME=
-INSTALL_MANPATH=/usr/local/man
+INSTALL_MANPATH=/data/data/com.termux/files/usr/share/man
 PCAP=-lpcap
 
 ARSOBJ = ars.o apd.o split.o rapd.o
@@ -50,20 +51,20 @@
 	$(RANLIB) $@
 
 hping3: byteorder.h $(OBJ)
-	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP)  -ltcl8.4 -lm -lpthread
+	$(CC) -o hping3 $(CCOPT) $(CPPFLAGS) $(LDFLAGS) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP)  -ltcl8.6 -lm -lpthread -landroid-shmem
 	@echo
-	./hping3 -v
+	
 	@echo "use \`make strip' to strip hping3 binary"
 	@echo "use \`make install' to install hping3"
 
 hping3-static: byteorder.h $(OBJ)
-	$(CC) -static -o hping3-static $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP)  -ltcl8.4 -lm -lpthread -ldl
+	$(CC) -static -o hping3-static $(CCOPT) $(CPPFLAGS) $(LDFLAGS) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP)  -ltcl8.6 -lm -lpthread -ldl
 
 byteorder.h:
 	./configure
 
 .c.o:
-	$(CC) -c $(CCOPT) $(DEBUG) $(COMPILE_TIME) $<
+	$(CC) -c $(CCOPT) $(CPPFLAGS) $(LDFLAGS) $(DEBUG) $(COMPILE_TIME) $<
 
 clean:
 	rm -rf hping3 *.o libars.a
@@ -72,10 +73,10 @@
 	rm -rf hping3 *.o byteorder byteorder.h systype.h Makefile libars.a .depend
 
 install: hping3
-	cp -f hping3 /usr/sbin/
-	chmod 755 /usr/sbin/hping3
-	ln -s /usr/sbin/hping3 /usr/sbin/hping
-	ln -s /usr/sbin/hping3 /usr/sbin/hping2
+	cp -f hping3 /data/data/com.termux/files/usr/bin/
+	chmod 755 /data/data/com.termux/files/usr/bin/hping3
+	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping
+	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping2
 	@if [ -d ${INSTALL_MANPATH}/man8 ]; then \
 		cp ./docs/hping3.8 ${INSTALL_MANPATH}/man8; \
 		chmod 644 ${INSTALL_MANPATH}/man8/hping3.8; \
diff -Naur hping3-20051105/Makefile.in hping3-3.0.0/Makefile.in
--- hping3-20051105/Makefile.in	2004-04-10 05:08:56.000000000 +0530
+++ hping3-3.0.0/Makefile.in	2018-02-22 18:27:09.000000000 +0530
@@ -6,9 +6,9 @@
 # $date: Sun Jul 25 17:56:15 MET DST 1999$ 
 # $rev: 3$ 
 
-CC= gcc
-AR=/usr/bin/ar
-RANLIB=/usr/bin/ranlib
+CC=@CC@
+AR=@AR@
+RANLIB=@RANLIB@
 CCOPT= -O2 -Wall @PCAP_INCLUDE@ @TCL_INC@ @USE_TCL@
 DEBUG= -g
 #uncomment the following if you need libpcap based build under linux
@@ -50,14 +50,14 @@
 	$(RANLIB) $@
 
 hping3: byteorder.h $(OBJ)
-	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@
+	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@
 	@echo
-	./hping3 -v
+
 	@echo "use \`make strip' to strip hping3 binary"
 	@echo "use \`make install' to install hping3"
 
 hping3-static: byteorder.h $(OBJ)
-	$(CC) -static -o hping3-static $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@ -ldl
+	$(CC) -static -o hping3-static $(CCOPT) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@ -ldl
 
 byteorder.h:
 	./configure
@@ -72,10 +72,10 @@
 	rm -rf hping3 *.o byteorder byteorder.h systype.h Makefile libars.a .depend
 
 install: hping3
-	cp -f hping3 /usr/sbin/
-	chmod 755 /usr/sbin/hping3
-	ln -s /usr/sbin/hping3 /usr/sbin/hping
-	ln -s /usr/sbin/hping3 /usr/sbin/hping2
+	cp -f hping3 /data/data/com.termux/files/usr/bin/
+	chmod 755 /data/data/com.termux/files/usr/bin/hping3
+	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping
+	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping2
 	@if [ -d ${INSTALL_MANPATH}/man8 ]; then \
 		cp ./docs/hping3.8 ${INSTALL_MANPATH}/man8; \
 		chmod 644 ${INSTALL_MANPATH}/man8/hping3.8; \
diff -Naur hping3-20051105/bytesex.h hping3-3.0.0/bytesex.h
--- hping3-20051105/bytesex.h	2003-08-31 22:53:48.000000000 +0530
+++ hping3-3.0.0/bytesex.h	2018-02-22 10:33:57.000000000 +0530
@@ -7,16 +7,10 @@
 #ifndef ARS_BYTESEX_H
 #define ARS_BYTESEX_H
 
-#if 	defined(__i386__) \
-	|| defined(__alpha__) \
-	|| (defined(__mips__) && (defined(MIPSEL) || defined (__MIPSEL__)))
+#include <endian.h>
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 #define BYTE_ORDER_LITTLE_ENDIAN
-#elif 	defined(__mc68000__) \
-	|| defined (__sparc__) \
-	|| defined (__sparc) \
-	|| defined (__PPC__) \
-	|| defined (__BIG_ENDIAN__) \
-	|| (defined(__mips__) && (defined(MIPSEB) || defined (__MIPSEB__)))
+#elif __BYTE_ORDER == __BIG_ENDIAN
 #define BYTE_ORDER_BIG_ENDIAN
 #else
 # error can not find the byte order for this architecture, fix bytesex.h
diff -Naur hping3-20051105/configure hping3-3.0.0/configure
--- hping3-20051105/configure	2004-06-04 12:09:10.000000000 +0530
+++ hping3-3.0.0/configure	2018-02-22 16:27:46.000000000 +0530
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/data/data/com.termux/files/usr/bin/sh
 
 show_help()
 {
@@ -15,14 +15,14 @@
 
 CC=${CC:=cc}
 
-echo build byteorder.c...
-$CC byteorder.c -o byteorder || exit 1
+#echo build byteorder.c...
+#$CC byteorder.c -o byteorder || exit 1
 
 INSTALL_MANPATH=`echo $MANPATH|cut -f1 -d:`
 if [ "$INSTALL_MANPATH" = "" ]; then
-	INSTALL_MANPATH="/usr/local/man"
+	INSTALL_MANPATH="$PREFIX/share/man"
 fi
-BYTEORDER=`./byteorder -m`
+BYTEORDER=__LITTLE_ENDIAN_BITFIELD
 
 echo create byteorder.h...
 cat > byteorder.h <<EOF
@@ -61,52 +61,52 @@
 #
 # TCL detection
 #
-for TCLPATH_TRY in "/usr/bin/" "/usr/local/bin/" "/bin/"
-do
-	for TCLVER_TRY in "8.4" "8.3" "8.2" "8.1" "8.0"
-	do
-		if [ -z $TCLSH ]
-		then
-			TCLSH_TRY=${TCLPATH_TRY}tclsh${TCLVER_TRY}
-			if [ -f $TCLSH_TRY ]
-			then
-				TCLSH=$TCLSH_TRY
-				echo "===> Found Tclsh in: $TCLSH"
-			fi
-		fi
-	done
-done
-if [ -f $TCLSH ]
-then
-	TCL_VER=`echo puts \\$tcl_version | $TCLSH -`
-	USE_TCL='-DUSE_TCL'
-	TCL_LIB="-ltcl${TCL_VER}"
-	if [ -e /usr/include/tcl${TCL_VER} ]
-	then
-		TCL_INC="-I/usr/include/tcl${TCL_VER}"
-	elif [ -e /usr/include/tcl.h ]
-	then
-		TCL_INC=""
-	elif [ -e /usr/local/include/tcl${TCL_VER} ]
-	then
-		TCL_INC="-I/usr/local/include/tcl${TCL_VER}"
-	else
-		USE_TCL=""
-		TCL_LIB=""
-		echo "==> WARNING: no Tcl header files found!"
-	fi
-fi
-if [ -n $USE_TCL ]
-then
-	LIBPOSTFIX=`ls -1 /usr/local/lib/ /usr/lib | grep 'libtcl[0-9]' | grep so | sed -e 's/\.so.*//g' -e 's/libtcl//g' | sort -r | head -1`
-	TCL_LIB="-ltcl${LIBPOSTFIX} -lm -lpthread"
-fi
+#for TCLPATH_TRY in "/data/data/com.termux/files/usr/bin/"
+#do
+#	for TCLVER_TRY in "8.6" "8.3" "8.2" "8.1" "8.0"
+#	do
+	#	if [ -z $TCLSH ]
+	#	then
+		#	TCLSH_TRY=${TCLPATH_TRY}tclsh${TCLVER_TRY}
+		#	if [ -f $TCLSH_TRY ]
+			#then
+			#	TCLSH=$TCLSH_TRY
+			#	echo "===> Found Tclsh in: $TCLSH"
+			#fi
+		#fi
+	#done
+#done
+#if [ -f $TCLSH ]
+#then
+	#TCL_VER=`echo puts \\$tcl_version | $TCLSH -`
+	#USE_TCL='-DUSE_TCL'
+#	TCL_LIB="-ltcl${TCL_VER}"
+	#if [ -e $PREFIX/include/tcl${TCL_VER} ]
+	#then
+	#	TCL_INC="-I$PREFIX/include/tcl${TCL_VER}"
+	#elif [ -e $PREFIX/include/tcl.h ]
+	#then
+		#TCL_INC=""
+	#elif [ -e $PREFIX/include/tcl${TCL_VER} ]
+	#then
+		#TCL_INC="-I$PREFIX/include/tcl${TCL_VER}"
+	#else
+		#USE_TCL=""
+		#TCL_LIB=""
+		#echo "==> WARNING: no Tcl header files found!"
+	#fi
+#fi
+#if [ -n $USE_TCL ]
+#then
+	#LIBPOSTFIX=`ls -1 $PREFIX/lib/ $PREFIX/lib | grep 'libtcl[0-9]' | grep so | sed -e 's/\.so.*//g' -e 's/libtcl//g' | sort -r | head -1`
+	#TCL_LIB="-ltcl${LIBPOSTFIX} -lm -lpthread -landroid-shmem"
+#fi
 
 #
 # configurable stuff
 #
-PCAP="PCAP=-lpcap"
-PCAP_INCLUDE=""
+PCAP="PCAP=-lpcap -landroid-shmem"
+PCAP_INCLUDE="-I/data/data/com.termux/files/usr/include"
 
 for ARG in $*; do
 	case "$ARG" in
@@ -143,6 +143,10 @@
 	-e "s^@TCL_INC@^$TCL_INC^g" \
 	-e "s^@TCL_VER@^$TCL_VER^g" \
 	-e "s^@TCL_LIB@^$TCL_LIB^g" \
+	-e "s^@AR@^$AR^g" \
+	-e "s^@CC@^$CC^g" \
+	-e "s^@LD@^$LD^g" \
+	-e "s^@RANLIB@^$RANLIB^g" \
 	<Makefile.in > Makefile
 
 #
diff -Naur hping3-20051105/libpcap_stuff.c hping3-3.0.0/libpcap_stuff.c
--- hping3-20051105/libpcap_stuff.c	2004-04-10 05:08:56.000000000 +0530
+++ hping3-3.0.0/libpcap_stuff.c	2018-02-11 18:37:24.000000000 +0530
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap/bpf.h>
 
 #include "globals.h"
 
diff -Naur hping3-20051105/scan.c hping3-3.0.0/scan.c
--- hping3-20051105/scan.c	2003-10-22 16:11:00.000000000 +0530
+++ hping3-3.0.0/scan.c	2018-02-11 20:07:34.000000000 +0530
@@ -20,7 +20,7 @@
 #include <sys/ipc.h>
 #endif
 #include <sys/shm.h>
-#include <sys/sem.h>
+#include <linux/sem.h>
 #include <unistd.h>
 #include <netdb.h>
 #include <sys/time.h>
@@ -34,7 +34,8 @@
 #include <fcntl.h>
 
 #if 0
-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
+#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED) || \
+defined(__ANDROID__)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
@@ -137,7 +138,7 @@
 {
 	int semid, sem_key;
 
-	if ((sem_key = ftok("/tmp/hpingscansem", 1)) == -1) {
+	if ((sem_key = ftok("/data/data/com.termux/files/usr/tmp/hpingscansem", 1)) == -1) {
 		perror("ftok");
 		exit(1);
 	}
diff -Naur hping3-20051105/script.c hping3-3.0.0/script.c
--- hping3-20051105/script.c	2004-05-29 12:18:13.000000000 +0530
+++ hping3-3.0.0/script.c	2018-02-11 20:10:14.000000000 +0530
@@ -24,7 +24,7 @@
 
 #include <sys/ioctl.h>
 #include <pcap.h>
-#include <net/bpf.h>
+#include <pcap/bpf.h>
 
 #include "release.h"
 #include "hping2.h"
