--- ../pam_cgfs.c.orig	2019-11-03 20:44:38.379704635 +0100
+++ ./src/lxc/pam/pam_cgfs.c	2019-11-03 20:46:09.764590541 +0100
@@ -1754,7 +1754,7 @@
 	oldv = *lastslash;
 	*lastslash = '\0';
 
-	fpath = must_make_path(path, "cpuset.cpus", NULL);
+	fpath = must_make_path(path, "cpus", NULL);
 	posscpus = read_file(fpath);
 	if (!posscpus) {
 		pam_cgfs_debug("Could not read file: %s\n", fpath);
@@ -1856,7 +1856,7 @@
 
 	free(fpath);
 
-	fpath = must_make_path(path, "cpuset.cpus", NULL);
+	fpath = must_make_path(path, "cpus", NULL);
 	ret = lxc_write_to_file(fpath, cpulist, strlen(cpulist), false, 0660);
 	if (ret < 0) {
 		pam_cgfs_debug("Could not write cpu list to: %s\n", fpath);
@@ -2025,8 +2025,8 @@
 	}
 
 	/* copy parent's settings */
-	if (!cg_copy_parent_file(cgpath, "cpuset.mems")) {
-		pam_cgfs_debug("%s", "Failed to copy \"cpuset.mems\" settings\n");
+	if (!cg_copy_parent_file(cgpath, "mems")) {
+		pam_cgfs_debug("%s", "Failed to copy \"mems\" settings\n");
 		free(cgpath);
 		free(clonechildrenpath);
 		return false;
